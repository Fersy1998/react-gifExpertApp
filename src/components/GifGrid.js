import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'

//{ useEffect, useState } from 'react'
import GifGridItem from './GifGridItem'
//import { getGifs } from '../helpers/GetGifs'
const GifGrid = ({category})=> {
  const {data:images, loading}=useFetchGifs(category);
  
  //console.log(state);
   // const [counter, setCounter] = useState(0)
    /*
  const [images, setImages] = useState([]);
  useEffect(() => {
    getGifs(category)
      .then(gifs => {
        console.log(gifs);
        setImages(...images, gifs);

      });
  }, [category])*/

    //<button onClick={()=>{setCounter(counter+1); console.log(counter)}} >getGifs</button>

    return (
      <>
      <h2>{category}</h2>
      {loading && <p>Loading</p>}
      <div className="GridGifs">
          
          {//en la linea de abajo cpm {...image} solo se envían las propiedades de la image
          images.map(image=> (<GifGridItem {...image} key={image.id}/>))
          }
      
      </div>
      
      
      
      </>
    )
    /*{loading ? 'cargando...': 'True'}
    <div className="GridGifs">
          
          {//en la linea de abajo cpm {...image} solo se envían las propiedades de la image
          images.map(image=> (<GifGridItem {...image} key={image.id}/>))
          }
      
      </div>*/
  }


export default GifGrid