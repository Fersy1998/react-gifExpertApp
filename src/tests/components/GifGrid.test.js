import React from 'react';
import {shallow} from 'enzyme';
import GifGrid from '../../components/GifGrid';
import '@testing-library/jest-dom';

//Controla la información que recibirá y arrojará useFetchGifs
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas con GifGrid.js', ()=>{
    let category="Pikachu";
    test('should  mach the snapshot', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const wrapper=shallow(< GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show and change the images when use the Custom Hook useFetchGifs.js', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper=shallow(< GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        //expect(wrapper).toMatchSnapshot;
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
})