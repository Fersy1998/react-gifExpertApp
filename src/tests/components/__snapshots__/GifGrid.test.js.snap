// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con GifGrid.js should  mach the snapshot 1`] = `
<Fragment>
  <h2>
    Pikachu
  </h2>
  <p>
    Loading
  </p>
  <div
    className="GridGifs"
  />
</Fragment>
`;

exports[`pruebas con GifGrid.js should show and change the images when use the Custom Hook useFetchGifs.js 1`] = `
<Fragment>
  <h2>
    Pikachu
  </h2>
  <div
    className="GridGifs"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;
